### PR TITLE
[PLT-1412] Updated search buttons on mobile GH-5261

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Updated search buttons on mobile

#### Ticket Link
GitHub issue: #5261 

#### Checklist
- [ x] Has UI changes

### CODE REVIEW

**Comment by jasonblais** [comment]
@doosp can you help doing the first test review?

**Comment by doosp** [comment]
@saturninoabril 

Thanks for another contribution.

I really like this feature. Awesome work!

I couldn't find any other issue when I tested.

@jasonblais @esethna would you please help review it? Let me know if there's any other test I might have missed. Thanks.

Test Summary:
1. Click the search button. No [x] button and cursor doesn't blink (#5284 is not merged yet)
2. The [x] button appears as soon as I start typing
3. [x] button is still there after the search because the search field is not blank
4. [<] button never disappear and I can go back to the room page anytime I want

**Comment by jasonblais** [comment]
Thanks @saturninoabril  and @doh5! Linda is helping test this PR as well

**Comment by lindalu** [comment]
Yes I am, and this looks great! Tested on desktop (OSX, both Chrome and FF, narrow view) and on Samsung Android phone in Chrome browser, mobile view.

1. Click Search button
2. On typing in the field, "x" button displays
3. Backspace to clear search field: "x" disappears and "Search" and cursor display in search field (as expected)
4. Type in field again: "x" returns
5. After search, text remains in field and "x" still displays
6. Click "x": text in search field is cleared
7. Back button is shown left of search field at all times
8. Click the Back button (with or without text in search field): Returns to center channel

Note: Verified that the "x" is easy to tap on mobile.

**Comment by jasonblais** [comment]
Thanks @doosp and @lindalu  with your help testing the PR!

@saturninoabril really appreciate your help here. This is a great improvement 👍

File: webapp/components/search_bar.jsx [hunk 1]
```diff
@@ -270,15 +274,6 @@ export default class SearchBar extends React.Component {
                 >
                     <span className='fa fa-angle-left'/>
                 </div>
-                <span
-                    className='search__clear'
-                    onClick={this.clearFocus}
-                >
-                    <FormattedMessage
-                        id='search_bar.cancel'
```
Comment (enahum) [inline webapp/components/search_bar.jsx]: is this string use elsewhere? if not can you please remove it from `webapp/i18n/en.json`?

**Comment by saturninoabril** [comment]
'search_bar.cancel' - this is not being used elsewhere. Aside from en, should I delete it as well from other locale?

File: webapp/components/search_bar.jsx (hunk 1 above)
Comment (saturninoabril) [inline webapp/components/search_bar.jsx]: 'search_bar.cancel' - this is not being used elsewhere. Aside from en, should I delete it as well from other locale?

Comment (enahum) [inline webapp/components/search_bar.jsx]: it should be fine if you remove it from `en.json` the process for translations will get rid of the other ones for you

**Comment by saturninoabril** [comment]
@enahum @coreyhulen deleted the 'search_bar.cancel' at en.json. thanks!

**Pull request merged**